ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java

```diff
@@ -96,7 +94,12 @@ public String getVersion() {
 
     @Override
     public boolean validate(Map<String, String> parameters) {
-        return parameters != null && parameters.containsKey(INPUT_FIELD) && parameters.get(INPUT_FIELD) != null;
+        if (parameters == null || parameters.isEmpty()) {
+            return false;
+        }
+        boolean argumentsFromInput = parameters.containsKey(INPUT_FIELD) && parameters.get(INPUT_FIELD) != null;
+        boolean argumentsFromParameters = parameters.containsKey(INDEX_FIELD) && parameters.containsKey(QUERY_FIELD);
```

Comment:
Better to check for bothel empty strings and null values
private boolean isValidValue(Map<String, String> params, String key) {
    String value = params.get(key);
    return value != null && !value.trim().isEmpty();
}
boolean argumentsFromInput = isValidValue(parameters, INPUT_FIELD);